fodmap-project/app/recipe-id/[id]/page.js

```diff
@@ -0,0 +1,193 @@
+"use client"
+
+import styled from "styled-components";
+import Navbar from "../../shared_components/navbar";
+import Image from "next/image";
+import bhaji from "../../../public/bhaji.jpg";
+
+const StyledWrapperDiv = styled.div`
+    display: flex;
+    background: #FCF8F3;
+    justify-content: center;
+
+    .content {
+        display: flex;
+        flex-wrap: wrap;
+        width: 900px;
+        margin: 30px 50px;
+    }
+    
+    .content > * {
+        padding: 30px;
+    }
+    
+    .recipe-image {
+        flex-basis: 40%;
+        border-radius: 10px;
+    }
+
+    .recipe-header {
+        display: flex;
+        width: 100%;
+        justify-content: space-between;
+    }
+
+    .recipe-highlights {
+        flex-basis: 40%;
+        display: flex;
+        flex-direction: column;
+        justify-content: space-between;
+    }
+
+    .recipe-highlights > * {
+        margin: 20px 0px;
+    }
+
+    h1 {
+        margin: 0px;
+    }
+    
+    h3 {
+        font-weight: 500;
+        margin-bottom: 0px;
+        font-style: italic;
+    }
+    
+    p {
+        margin-top: 0px;
+    }
+
+
+    .recipe-label-container {
+        display: flex;
+        flex-wrap: wrap;       
+    }
+
+
+
+    .recipe-ingredients, .recipe-method {
+        flex: 1 1 100%;
+    }
+
+`
+
+const StyledLabel = styled.button`
+    color: #FCF8F3;
+    background-color: #698474;
+    border-radius: 5px;
+    padding: 5px 10px;
+    margin: 7.5px
+`
+
+
+
+function RecipeLabel(props) {
+    return (
+        props.label.map(item => {
+            return (
+                <StyledLabel>{item}</StyledLabel>
+            )
+        })
+    )
+}
+
+function ListRecipeDetails(props) {
+    return (
+        <>
+        <h2>{props.heading}</h2>
+        {
+            props.steps.map((step, index) => {
```

Comment:
it looks like nested components. 
it's better to move the function inside map to a separate function.

Reply:
Did you mean for me to create a new component with the map function below?

Reply:
I've created two separate functions for the mapping of the ingredients and methods as per our discussion.